code401-challenges/src/main/java/code401/challenges/treeIntersection/TreeIntersection.java

```diff
@@ -0,0 +1,50 @@
+package code401.challenges.treeIntersection;
+
+import code401.challenges.tree.Node;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class TreeIntersection {
+
+        public ArrayList<Integer> tree_intersection(Node root1, Node root2) {
+            ArrayList<Integer> duplicateArrList = new ArrayList<>();
+            if(root1 == null){
+                return duplicateArrList;
+            }
+            if(root2 == null){
+                return duplicateArrList;
+            }
+
+            HashMap<Integer, Integer> duplicateHash = new HashMap<>();
+            int dummyCnt = 0;
+            hashFirstTree(root1, duplicateHash, dummyCnt);
+            hashSecondTree(root2, duplicateHash, dummyCnt, duplicateArrList);
+            return duplicateArrList;
+        }
+
+    private void hashFirstTree(Node<Integer> node, HashMap<Integer, Integer> duplicateHash, int dummyCnt){
+            if(node == null){
+                return;
+            }else{
+                dummyCnt++;
+                if(duplicateHash.get(node.getValue()) == null){
+                    duplicateHash.put(node.getValue(), dummyCnt);
+                }
+                hashFirstTree(node.getLeft(), duplicateHash, dummyCnt);
+                hashFirstTree(node.getRight(), duplicateHash, dummyCnt);
+            }
+    }
+
+    private void hashSecondTree(Node<Integer> node, HashMap<Integer, Integer> duplicateHash, int dummyCnt, ArrayList<Integer> duplicateArrList){
```

Comment:
This method, since it is doing the same thing as the previous method just on a different tree, make your code not DRY

Reply:
It is doing a different thing than the previous method